libraries/ms-helper/src/persist.c

```diff
@@ -0,0 +1,158 @@
+#include "persist.h"
+#include <inttypes.h>
+#include "log.h"
+#include "soft_timer.h"
+// The persistance layer allocates one page of flash so we can erase the entire page when full.
```

Comment:
You might want to mention that there can be multiple active persistence instances. I didn't realize that at first.

Reply:
Honestly, I didn't intend for that to be possible but it seems like it is. :P
In that case, should I make the flash page a user parameter?

Reply:
Yeah that might be nice. It would be helpful for things like separate module configs to be able to persist and it reduces the need to pipe data around into some universal persist struct.